src/pharmpy/tools/estmethod/algorithms.py

```diff
@@ -171,31 +240,37 @@ def _create_candidate_model(
     while len(model.estimation_steps) > 0:
         model = remove_estimation_step(model, 0)
 
-    model = add_estimation_step(model, **est_settings)
-    model = add_estimation_step(model, **eval_settings)
+    if only_evaluation:
+        model = add_estimation_step(model, **eval_settings)
+    elif add_eval_after_est:
+        model = add_estimation_step(model, **est_settings)
+        model = add_estimation_step(model, **eval_settings)
+    else:
+        model = add_estimation_step(model, **est_settings)
+
     if solver:
         model = set_ode_solver(model, solver)
     return model
 
 
-def _create_est_settings(method, only_evaluation=False):
+def _create_est_settings(method, parameter_uncertainty_method=None, add_eval_after_est=False):
     est_settings = {
         'method': method,
         'interaction': True,
         'laplace': False,
         'auto': True,
         'keep_every_nth_iter': 10,
-        'parameter_uncertainty_method': None,
+        'maximum_evaluations': 9999,
```

Comment:
How come this setting is added?

Reply:
Because the idea is that `est_settings` should always perform estimation, so MAXEVAL>0. If an evaluation is requested (which is the case for MAXEVAL=0), then we use the `eval_settings`. Currently only IMP evaluation is implemented, and not FOCE evaluation. Previously, evaluation was done both FOCE and then IMP in the same model, which makes no sense to me. I suggest we implement FOCE evaluation in a later PR.

Reply:
And `parameter_uncertainty_method` is allowed to be something else than `None` since we want to be able to just update that and not be forced to add an `IMP` evaluation step if we want to update the `parameter_uncertainty_method`

Reply:
FOCE evaluation is supported here: 
https://github.com/pharmpy/pharmpy/blob/3008997a229bd0f492b1ce08775d544a5176bba8/src/pharmpy/model/external/nonmem/update.py#L1514-L1520
And tested for her:
https://github.com/pharmpy/pharmpy/blob/3008997a229bd0f492b1ce08775d544a5176bba8/src/pharmpy/model/external/nonmem/update.py#L1531-L1532
Basically the idea is that `evaluation=True` is independent of estimation method.

Reply:
`maximum_evaluations` is allowed to be `None` meaning whatever is the default (a "fair" amount according to the NONMEM userguide)

Reply:
Sure, FOCE evaluation is supported elsewhere in the Pharmpy code base, but in the `estmethod` tool it was never genuinely supported since an `IMP EONLY=1` step always followed.